includes/class-sensei-course.php

```diff
@@ -145,17 +145,87 @@ public function __construct() {
 
 		add_action( 'template_redirect', [ $this, 'setup_single_course_page' ] );
 		add_action( 'sensei_loaded', [ $this, 'add_legacy_course_hooks' ] );
+
+		// Add custom navigation.
+		add_action( 'in_admin_header', [ $this, 'add_custom_navigation' ] );
+		add_action( 'submenu_file', [ $this, 'highlight_menu_item' ] );
+	}
+
+	/**
+	 * Highlight the menu item for the course pages.
+	 *
+	 * @since 4.0.0
+	 * @access private
+	 *
+	 * @param string $submenu_file
+	 *
+	 * @return string
+	 */
+	public function highlight_menu_item( $submenu_file ) {
+		$screen = get_current_screen();
+
+		if ( $screen && in_array( $screen->id, [ 'edit-course', 'edit-course-category', 'course_page_course-order' ], true ) ) {
+			$submenu_file = 'edit.php?post_type=course';
+		}
+
+		return $submenu_file;
 	}
 
 	/**
-	 * Register and enqueue scripts that are needed in the backend.
+	 * Add custom navigation to the admin pages.
+	 *
+	 * @since 4.0.0
+	 * @access private
+	 */
+	public function add_custom_navigation() {
+		$screen = get_current_screen();
+		if ( ! $screen ) {
+			return;
+		}
+		if ( in_array( $screen->id, [ 'edit-course', 'edit-course-category' ], true ) ) {
+			$this->display_courses_navigation( $screen );
+		}
+	}
+
+	/**
+	 * Display the courses' navigation.
+	 *
+	 * @param WP_Screen $screen
+	 */
+	private function display_courses_navigation( WP_Screen $screen ) {
+		?>
+		<div id="sensei-custom-navigation" class="sensei-custom-navigation">
+			<div class="sensei-custom-navigation__main">
+				<div class="sensei-custom-navigation__main__title">
```

Comment:
To prevent these selectors from getting too long, there's no need to preserve the hierarchy in the naming. So for example, `sensei-custom-navigation__main__title` could be `sensei-custom-navigation__title`. We normally wouldn't have more than one set of `__` in a given selector. Same thing would apply for the other CSS class names in this file.

Reply:
Fixed here: https://github.com/Automattic/sensei/pull/4672/commits/08e1fa3d2e76062ef6abe8c958ba000f3131b54e